src/main/java/plugily/projects/villagedefense/kits/premium/TornadoKit.java

```diff
@@ -85,100 +93,152 @@ public Material getMaterial() {
-          VersionUtils.sendParticles("CLOUD", null, location.clone().add(x, y, z), 1, 0, 0, 0);
+        Vector velocityVec = vector.multiply(2).setY(0).add(new Vector(0, 1, 0));
+        if (VersionUtils.isPaper() && (vector.getX() > 4.0 || vector.getZ() > 4.0)) {
+          velocityVec = vector.setX(2.0).setZ(1.0); // Paper's sh*t
         }
+        ((LivingEntity) entity).damage(5.0);
+        entity.setVelocity(velocityVec);
       }
-      pushNearbyEnemies();
-      setLocation(location.add(vector.getX() / (3 + Math.random() / 2), 0, vector.getZ() / (3 + Math.random() / 2)));
-
-      angle += 50;
     }
+    return pierce;
+  }
 
-    private void pushNearbyEnemies() {
-      for(Entity entity : location.getWorld().getNearbyEntities(location, 2, 2, 2)) {
-        if(CreatureUtils.isEnemy(entity)) {
-          entities++;
+  private void createMonsoon(User user) {
+    Player player = user.getPlayer();
+    final int spellTime = getMonsoonSpellTime((Arena) user.getArena());
+    int cooldown = getKitsConfig().getInt("Kit-Cooldown.Tornado.Monsoon", 20);
+    user.setCooldown("tornado_monsoon", cooldown);
+    user.setCooldown("tornado_monsoon_running", spellTime);
 
-          Vector velocityVec = vector.multiply(2).setY(0).add(new Vector(0, 1, 0));
-          if(VersionUtils.isPaper() && (vector.getX() > 4.0 || vector.getZ() > 4.0)) {
-            velocityVec = vector.setX(2.0).setZ(1.0); // Paper's sh*t
-          }
+    List<String> messages = getPlugin().getLanguageManager().getLanguageListFromKey("KIT_CONTENT_TORNADO_GAME_ITEM_MONSOON_ACTIVE_ACTION_BAR");
+    new BukkitRunnable() {
+      int spellTick = 0;
+      int messageIndex = 0;
 
-          entity.setVelocity(velocityVec);
+      @Override
+      public void run() {
+        XParticle.circle(3.5, 28, ParticleDisplay.simple(player.getLocation().add(0, 0.5, 0), XParticle.getParticle("CLOUD")));
+
+        if (spellTick % 20 == 0) {
+          for (Entity en : player.getNearbyEntities(3.5, 3.5, 3.5)) {
+            if (!CreatureUtils.isEnemy(en) || en.equals(player)) {
+              continue;
+            }
+            LivingEntity entity = (LivingEntity) en;
+            //damage for 0 to knock it back, todo vector push here
+            entity.damage(0, user.getPlayer());
+            entity.addPotionEffect(new PotionEffect(PotionEffectType.SLOW, 20 * 9999, 2, false, true));
+          }
+        }
+        if (spellTick % 10 == 0) {
+          VersionUtils.sendActionBar(player, messages.get(messageIndex)
+              .replace("%number%", String.valueOf(user.getCooldown("tornado_monsoon_running"))));
+          messageIndex++;
+          if (messageIndex > messages.size() - 1) {
+            messageIndex = 0;
+          }
+        }
+        if (spellTick >= 20 * spellTime || !getPlugin().getArenaRegistry().isInArena(user.getPlayer()) || user.isSpectator()) {
+          //reset action bar
+          VersionUtils.sendActionBar(player, "");
```

Comment:
is it needed to reset actionbar really or would it be okay to just end actionbar?

Reply:
no, it looks better if we send if empty rather than disappear with fade

Reply:
ok